mantidproject/mantid
# Fit Script Generator - Implement Edit Local Parameter dialog

**Description of work.**
This PR implements the Edit Local Parameter dialog for the Fit Script Generator interface. It is now possible to select a local parameter using the `...` button in the function tree and edit the values/ties/fixes/constraints for this parameter across all the domains that it exists in.

**To test:**
1. Comment out this line
https://github.com/mantidproject/mantid/blob/44daa7417226a56f8d1896ea6b2f2b1bc576a5d1/scripts/Muon/GUI/Common/fitting_tab_widget/fitting_tab_view.py#L52
2. Go to Muon Analysis interface
3. click `Load Current Run`
4. Go to Fitting tab and click `Fit Generator`
5. Click `Add Workspaces`. Select the first workspace and click add.
6. Right click the function browser and add a `FlatBackground` and then `ExpDecay`
7. Select a parameter and click the `...`. The Edit Local Parameter dialog should open.
8. Try out fixing, tying and constraining the parameter. Click `Ok` each time and check the changes have been applied within the function tree. 
9. Opening the dialog should also load in the current fixes/ties/constraints seen in the function tree.

Fixes #30684 

*This does not require release notes* because **the Fit Script Generator is inaccessible to users at the moment.**

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
